# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/PendingStyles.cpp
# Commit: dfd4ec691465
# Full Hash: dfd4ec69146560b6de52edbc4acdb49762daf38c
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2023-01-17 16:13:02
# Regressor Bug: 1807829
# File Overlap Count: 2
# Description:
#   Bug 1807829 - Make `HTMLEditor::CreateStyleForInsertText` use same logic as `HTMLEditor::SetInlinePropertiesAsSubAction` to apply new style r=m_kato
#   
#   The problem in Yahoo mail is, when you apply new font size to:
#   ```
#   <div><font size="7">{}<br></font></div>
# ==============================================================================

diff -r 0f2913c81558 -r dfd4ec691465 editor/libeditor/PendingStyles.cpp
--- a/editor/libeditor/PendingStyles.cpp	Mon Jan 16 23:35:34 2023 +0000
+++ b/editor/libeditor/PendingStyles.cpp	Mon Jan 16 23:42:27 2023 +0000
@@ -408,6 +408,21 @@
       aHTMLProperty, aAttribute, u""_ns, aSpecifiedStyle));
 }
 
+void PendingStyles::TakeAllPreservedStyles(
+    nsTArray<EditorInlineStyleAndValue>& aOutStylesAndValues) {
+  aOutStylesAndValues.SetCapacity(aOutStylesAndValues.Length() +
+                                  mPreservingStyles.Length());
+  for (UniquePtr<PendingStyle> preservedStyle = TakePreservedStyle();
+       preservedStyle; preservedStyle = TakePreservedStyle()) {
+    aOutStylesAndValues.AppendElement(
+        preservedStyle->GetAttribute()
+            ? EditorInlineStyleAndValue(
+                  *preservedStyle->GetTag(), *preservedStyle->GetAttribute(),
+                  preservedStyle->AttributeValueOrCSSValueRef())
+            : EditorInlineStyleAndValue(*preservedStyle->GetTag()));
+  }
+}
+
 /**
  * TakeRelativeFontSize() hands back relative font value, which is then
  * cleared out.